src/Components/Popover.tsx

```diff
@@ -0,0 +1,154 @@
+    React.HTMLProps<HTMLDivElement>
+> =
+  styled.div
+
+const Container = Positionable`
+  position: fixed;
+  top: ${props => props.adjustTop};
+  top: ${props => props.position && props.position.top}px;
+  left: ${props => props.position && props.position.left}px;
+  width: ${props => props.position && props.position.width};
+`
+
+const Tip = Positionable`
+  position: absolute;
+  display: ${props => (props.hide ? "none" : "block")};
+  visibility: ${props => (props.hide ? "hidden" : "visible")};
+  animation: ${props => (props.hide ? fadeOut : fadeIn)} 1s linear;
+  z-index: 1000;
+  // align-items: center;
+  // flex-direction: column;
+`
+
+const Arrow = Positionable`
+  width: 0;
+  height: 0;
+  border-bottom: 10px solid #000;
+  border-left: 10px solid transparent;
+  border-right: 10px solid transparent;
+  position: absolute;
+  left: ${props => props.position && props.position.width / 2 - 10}px;
+`
+// or:
+// width: 12px;
+// height: 12px;
+// transform: rotate(-45deg);
+// background: #000;
+
+const Message = Positionable`
+  ${/*props => (props.position && console.log(props.position.width, props.ownWidth)) || */ ""}
+  position: absolute;
+  top: 10px;
+  left: ${props =>
+    props.position && props.position.width / 2 - props.ownWidth / 2}px;
+  padding: 13px 15px;
+  background: #000;
+  color: #fff;
+  font-family: "Courier New", Courier, monospace;
+  cursor: default;
+  & span {
+    cursor: pointer;
+  }
+`
+
+export class Popover extends React.Component<Props, any> {
+  public messageRef
+
+  static defaultProps = {
+    hide: true,
+    maxWidth: "200px",
+    dismissOnClick: true,
+    // dismissTimeout: null
+  }
+
+  constructor(props) {
+    super(props)
+    const message = this.getMessage(props)
+    const hide = props.hide === true || !message
+    this.state = {
+      hide,
+      message,
+      anchorPosition: this.getAnchorPosition(props.anchorRef),
+    }
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
+    const newMessage = this.getMessage(nextProps)
+    const hide = !newMessage || newMessage === this.state.message
+    this.setState({
+      hide,
+      message: newMessage,
+      anchorPosition: this.getAnchorPosition(nextProps.anchorRef),
+      messageWidth: this.getMessageWidth(),
+    })
+  }
+
+  getAnchorPosition: (ref: any) => PositionProps = ref => {
+    if (ref) {
+      const anchor = ReactDOM.findDOMNode(ref)
+      const rect = anchor.getBoundingClientRect()
+      const { top, left, width, height } = rect
+
+      return { width, left, top: top + height }
+    }
+  }
+
+  getMessageWidth: (fb?: number) => number = (fallback = 50) => {
+    if (this.messageRef) {
+      const el = ReactDOM.findDOMNode(this.messageRef)
+      return el.getBoundingClientRect().width || fallback
+    }
+    return fallback
+  }
+
+  getMessage = props =>
+    (typeof props.children === "function"
+      ? props.children()
+      : props.children) || ""
```

Comment:
This is a bit odd; what is going on here and how is it used in context? I would also recommend always using `{ }` for functions, and include `return` since implicit return gets future-buggy and somewhat clever when its spread across more than one line.

Reply:
:+1: maybe not necessary anyways since i'm not currently passing any functions to my children. I think prettier made it look like this though.